firmware/ventilator-controller-stm32/Core/Src/Pufferfish/Driver/I2C/LTC4015/Device.cpp

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2020, the Pez Globo team and the Pufferfish project contributors
+ *
+ *      Author: Rohan Purohit
+ *
+ *  A driver for LTC4015 buck battery charger
+ */
+
+#include "Pufferfish/Driver/I2C/LTC4015/Device.h"
+
+#include "Pufferfish/HAL/Interfaces/Endian.h"
+#include "Pufferfish/Util/Endian.h"
+
+namespace Pufferfish::Driver::I2C::LTC4015 {
+
+I2CDeviceStatus Device::read_charging_status(bool &charging_status) {
+  std::array<uint8_t, sizeof(uint16_t)> buffer{};
+  I2CDeviceStatus ret = i2cdevice_.read(static_cast<uint16_t>(Command::system_status), buffer);
+  if (ret != I2CDeviceStatus::ok) {
+    return ret;
+  }
+
+  charging_status = false;
+  uint16_t system_status = 0;
+  Util::read_bigend(buffer.data(), system_status);
+  if ((system_status & static_cast<uint16_t>(Mask::charger_enabled)) != 0) {
+    charging_status = true;
+  }
+  return I2CDeviceStatus::ok;
+}
+
+I2CDeviceStatus Device::read_battery_voltage(uint16_t &v_bat) {
+  std::array<uint8_t, sizeof(uint16_t)> buffer{};
+  I2CDeviceStatus ret = i2cdevice_.read(static_cast<uint16_t>(Command::vbat_addr), buffer);
+  if (ret != I2CDeviceStatus::ok) {
+    return ret;
+  }
+
+  uint16_t vbat_value = 0;
+  Util::read_bigend(buffer.data(), vbat_value);
+  // NOLINTNEXTLINE(readability-magic-numbers)
+  v_bat = vbat_value * 128.176;  // Vbat = [VBAT] • 128.176µV for lead-acid
```

Comment:
All magic numbers for this driver should be defined as static constants, either private in the class declaration of `Device` (if the numbers aren't needed elsewhere) or in `Types.h` (if the numbers are needed elsewhere).

Reply:
It would be great if you could add code comments to these formulas (and/or to the definitions of static constants for magic numbers) indicating where to find these formulas in the datasheet (e.g. a URL and page number)

Reply:
Yep! added link for datasheet along with page number, they are only required now in `Device` so i have added it there.